### PR TITLE
simplewallet: add checks against service node list state to CLI for stake commands

### PR DESCRIPTION
**Pull request by jcktm**
This depends on get service node list state. Merge the other one first

### CODE REVIEW

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -5070,6 +5064,67 @@ bool simple_wallet::stake(const std::vector<std::string> &args_)
...
   }
 
+  try
+  {
+    const auto& response = m_wallet->get_service_node_list();
+    bool found = false;
+    bool full = false;
+    uint64_t can_contrib_total = 0;
+    uint64_t must_contrib_total = 0;
+    for (const auto& snode_info : response.service_node_states)
```
Comment (Doy-lee) [inline src/simplewallet/simplewallet.cpp]: Please add the squiggly braces for multi-line for loops.

File: src/simplewallet/simplewallet.cpp [hunk 2]
```diff
@@ -5070,6 +5064,67 @@ bool simple_wallet::stake(const std::vector<std::string> &args_)
...
+    {
+      fail_msg_writer() << tr("You may not contribute any more to this service node");
+      return true;
+    }
+    if (amount > can_contrib_total)
+    {
+      fail_msg_writer() << tr("You may only contribute up to ") << print_money(can_contrib_total) << tr(" more loki to this service node");
+      return true;
+    }
+    if (amount < must_contrib_total)
```
Comment (Doy-lee) [inline src/simplewallet/simplewallet.cpp]: This feels like it should just be a printed message, not a fail case. Use case, person has 2 wallets and wants to contribute funds from each, they shouldn't be required to consolidate their funds.

File: src/simplewallet/simplewallet.cpp [hunk 3]
```diff
@@ -4830,6 +4845,25 @@ bool simple_wallet::register_service_node(const std::vector<std::string> &args_)
     return true;
   }
 
+  try
+  {
+    const auto& response = m_wallet->get_service_node_list();
+    bool found = false;
+    for (const auto& snode_info : response.service_node_states)
+      if (string_to_public_key(snode_info.service_node_pubkey) == service_node_key)
+        found = true;
```
Comment (Doy-lee) [inline src/simplewallet/simplewallet.cpp]: Early break out.

But not necessary. I've updated the RPC call to accept a service node key for it to look for so I'll update this after the updated RPC call is added in.

File: src/simplewallet/simplewallet.cpp (hunk 1 above)
Comment (jcktm) [reply src/simplewallet/simplewallet.cpp]: nice, fixed

File: src/simplewallet/simplewallet.cpp (hunk 2 above)
Comment (jcktm) [reply src/simplewallet/simplewallet.cpp]: Yes, you're right. Good idea.

**Pull request merged**